### PR TITLE
Worker utils changes to always create a config in TDS.

### LINKED ISSUE #43: Create default model config for equation to AMR
When a `model_id` is provided, we [bypass adding a default config here](https://github.com/DARPA-ASKEM/knowledge-middleware/blob/main/worker/utils.py#L46-L79) but apparently this is breaking Terarium so we _do_ need to add the default config.

This should be simple enough by unindenting out of the `else` block [these lines](https://github.com/DARPA-ASKEM/knowledge-middleware/blob/main/worker/utils.py#L59-L75). We should bump up that logger statement around creating the model to where it belongs (in the `else` block).

Test this by adding sending these [equations](https://github.com/DARPA-ASKEM/knowledge-middleware/blob/main/tests/scenarios/basic/equations.txt) to the endpoint AFTER manually creating a placeholder model in TDS while providing that placeholder model's ID. Ensure that the placeholder model is appropriately updated in TDS and that a model configuration is created for that model.

Let's also make sure that our [test](https://github.com/DARPA-ASKEM/knowledge-middleware/blob/main/tests/test_e2e.py#L152-L199) covers this case since it's more important to the HMI than the case where no model yet exists (the HMI always creates a placeholder first). @fivegrant do you have any suggestions on how to efficiently test both modes?

### PR DESCRIPTION
**Pull request by Sorrento110**
# Description

* Moved the configuration creation to always happen at the end of the `put_amr_to_tds` process.
* Changed how configuration name and description are initialized to make it more consistent (since they are required by TDS).

Resolves #43

### CODE REVIEW

File: tests/test_e2e.py [hunk 1]
```diff
@@ -181,26 +211,48 @@ def test_equations_to_amr(context_dir, http_mock, client, worker, file_storage):
...
+    assert (
+        status_response.json().get("status") == "finished"
+    ), f"The RQ job failed.\n{job.latest_result().exc_string}"
 
     assert (
-            amr_instance.is_valid()
+        amr_instance.is_valid()
     ), f"AMR failed to validate to its provided schema: {amr_instance.validation_error}"
 
+    # Assertions for predefined model id.
```
Comment (fivegrant) [inline tests/test_e2e.py:241]: How does this assert the predefined model id?

Comment (Sorrento110) [reply tests/test_e2e.py]: Addressed in the newest commit with assertions to test the configurations.

File: worker/utils.py [hunk 2]
```diff
@@ -47,36 +47,45 @@ def put_amr_to_tds(amr_payload, name=None, description=None, model_id=None):
         tds_models = f"{TDS_API}/models/{model_id}"
```
[the remark this answers is not in the document]
Comment (Sorrento110) [reply worker/utils.py:47]: Changed in the newest commit.

File: worker/utils.py [hunk 3]
```diff
@@ -47,36 +47,45 @@ def put_amr_to_tds(amr_payload, name=None, description=None, model_id=None):
...
         model_response = requests.put(tds_models, json=amr_payload, headers=headers)
         if model_response.status_code != 200:
-            raise Exception(f"Cannot update model {model_id} in TDS with payload:\n\n {amr_payload}")
+            raise Exception(
+                f"Cannot update model {model_id} in TDS with payload:\n\n {amr_payload}"
+            )
         logger.info(f"Updated model in TDS with id {model_id}")
-        return {"model_id": model_id}
+        # return {"model_id": model_id}
```
[the remark this answers is not in the document]
Comment (Sorrento110) [reply worker/utils.py]: Removed in the newest commit.

File: worker/utils.py [hunk 4]
```diff
@@ -47,36 +47,45 @@ def put_amr_to_tds(amr_payload, name=None, description=None, model_id=None):
...
-            tds_model_configurations,
-            data=json.dumps(configuration_payload, default=str),
-            headers=headers,
-        )
+        logger.info(f"Created model in TDS with id {model_id}")
 
-        config_id = config_response.json().get("id")
+    # Create TDS model configuration
+    tds_model_configurations = TDS_API + "/model_configurations"
+    configuration_payload = {
```
[the remark this answers is not in the document]
Comment (Sorrento110) [reply worker/utils.py:66]: Changed in the newest commit to this solution to put the `header` values first.

File: worker/utils.py [hunk 5]
```diff
@@ -115,7 +124,14 @@ def put_document_extraction_to_tds(
 
 
 def put_code_extraction_to_tds(
-    code_id, name, description, filename, extractions=None, text=None, model_id=None, code_language=None
+    code_id,
+    name,
+    description,
+    filename,
```
[the remark this answers is not in the document]
Comment (Sorrento110) [reply worker/utils.py:126]: These are just linting changes as discussed in the huddle.

File: tests/test_e2e.py [hunk 6]
```diff
@@ -160,14 +180,24 @@ def test_equations_to_amr(context_dir, http_mock, client, worker, file_storage):
         "description": "test description",
     }
 
+    query_params_config_test = {
+        "equation_type": "latex",
+        "model": "petrinet",
+        "model_id": "test_id",
```
[the remark this answers is not in the document]
Comment (Sorrento110) [reply tests/test_e2e.py]: Changed this ID to `test2` for clarity in the newest commit, this distinction is for the second test case.

**Review by brandomr — APPROVED** [review]

**Review by fivegrant — APPROVED** [review]
lgtm

**Pull request merged**